R/facet-wrap.R

```diff
@@ -257,227 +257,203 @@ FacetWrap <- ggproto("FacetWrap", Facet,
-        if (params$strip.position == "top" &&
-            !inside &&
-            any(!vapply(x_axes, is.zero, logical(1))) &&
-            !params$free$x) {
-          cli::cli_warn("Suppressing axis rendering when {.code strip.position = \"top\"} and {.code strip.placement == \"outside\"}")
-        } else {
-          axis_mat_x_top[pos] <- x_axes
-        }
-      }
-      empty_right <- t(apply(empties, 1, function(x) c(diff(x) == 1, FALSE)))
-      if (any(empty_right)) {
-        pos <- which(empty_right)
-        panel_loc <- data_frame0(
-          ROW = row_ind[pos],
-          COL = col_ind[pos],
-          .size = length(pos)
-        )
-        panels <- vec_match(panel_loc, layout[, c("ROW", "COL")])
-        y_axes <- axes$y$right[y_axis_order[panels]]
-        if (params$strip.position == "right" &&
-            !inside &&
-            any(!vapply(y_axes, is.zero, logical(1))) &&
-            !params$free$y) {
-          cli::cli_warn("Suppressing axis rendering when {.code strip.position = \"right\"} and {.code strip.placement == \"outside\"}")
-        } else {
-          axis_mat_y_right[pos] <- y_axes
-        }
-      }
-      empty_left <- t(apply(empties, 1, function(x) c(FALSE, diff(x) == -1)))
-      if (any(empty_left)) {
-        pos <- which(empty_left)
-        panel_loc <- data_frame0(
-          ROW = row_ind[pos],
-          COL = col_ind[pos],
-          .size = length(pos)
-        )
-        panels <- vec_match(panel_loc, layout[, c("ROW", "COL")])
-        y_axes <- axes$y$left[y_axis_order[panels]]
-        if (params$strip.position == "left" &&
-            !inside &&
-            any(!vapply(y_axes, is.zero, logical(1))) &&
-            !params$free$y) {
-          cli::cli_warn("Suppressing axis rendering when {.code strip.position = \"left\"} and {.code strip.placement == \"outside\"}")
-        } else {
-          axis_mat_y_left[pos] <- y_axes
-        }
-      }
-    }
-    panel_table <- weave_axes(
-      panel_table,
-      axes = list(
-        top  = axis_mat_x_top,  bottom = axis_mat_x_bottom,
-        left = axis_mat_y_left, right  = axis_mat_y_right
-      ),
-      empty = empties
-    )
-    axis_size   <- panel_table$sizes
-    panel_table <- panel_table$panels
-
-    strip_padding <- convertUnit(calc_element("strip.switch.pad.wrap", theme), "cm")
-    strip_name <- paste0("strip-", substr(params$strip.position, 1, 1))
-    strip_mat <- empty_table
-    strip_mat[panel_pos] <- unlist(unname(strips), recursive = FALSE)[[params$strip.position]]
-    if (params$strip.position %in% c("top", "bottom")) {
-      inside_x <- (theme$strip.placement.x %||% theme$strip.placement %||% "inside") == "inside"
-      if (params$strip.position == "top") {
-        placement <- if (inside_x) -1 else -2
-        strip_pad <- axis_size$top
+  draw_panels = function(self, panels, layout, x_scales, y_scales, ranges, coord, data, theme, params) {
+    if (inherits(coord, "CoordFlip")) {
```

Comment:
If failed to find a smarter solution to working out this interaction with CoordFlip, so FacetWrap does this first, then calls the parent method.